scripts/w3c.js

```diff
@@ -1,354 +1,647 @@
-    "https://www.w3.org/TR/REC-DSig-label/": "https://www.w3.org/TR/DSig-label/",
-    "https://www.w3.org/TR/REC-MathML": "https://www.w3.org/TR/MathML/",
-    "https://www.w3.org/TR/REC-PICS-labels": "https://www.w3.org/TR/PICS-labels/",
-    "https://www.w3.org/TR/REC-PICS-services": "https://www.w3.org/TR/PICS-services/",
-    "https://www.w3.org/TR/REC-PICSRules": "https://www.w3.org/TR/PICSRules/",
-    "https://www.w3.org/TR/REC-WebCGM": "https://www.w3.org/TR/WebCGM/",
-    "https://www.w3.org/TR/REC-png": "https://www.w3.org/TR/PNG/",
-    "https://www.w3.org/TR/REC-rdf-syntax": "https://www.w3.org/TR/rdf-syntax-grammar/",
-    "https://www.w3.org/TR/REC-smil/": "https://www.w3.org/TR/SMIL/",
-    "https://www.w3.org/TR/REC-xml-names": "https://www.w3.org/TR/xml-names/",
-    "https://www.w3.org/TR/REC-xml": "https://www.w3.org/TR/xml/",
-    "https://www.w3.org/TR/xml-events": "https://www.w3.org/TR/xml-events2/",
-    "https://www.w3.org/TR/2001/WD-xhtml1-20011004/": "https://www.w3.org/TR/xhtml1/",
+// Record the entries that used to be aliases and that get turned into concrete
+// entries, so that we may reverse the alias link in the end.
+const aliasesToInvert = {};
+
+// Record the number of requests sent to the W3C API
+let requestsToW3CApi = 0;
+
+/**
+ * Specref uses abbreviations for W3C statuses.
+ *
+ * TODO: It would probably be better to use longer forms throughout. That would
+ * require updating all statuses in refs/w3c.json at once, and making sure that
+ * consumers are aware of the change first...
+ *
+ * Note that the W3C API considers that the status of a Retired spec is
+ * "Retired", while Specref records that information on the side and uses the
+ * status of the spec before it got retired.
+ */
+const STATUSES = {
+    'First Public Working Draft': 'FPWD',
+    'Working Draft': 'WD',
+    'Last Call Working Draft': 'LCWD',
+    'Candidate Recommendation': 'CR',
+    'Candidate Recommendation Draft': 'CRD',
+    'Candidate Recommendation Snapshot': 'CR',
+    'Proposed Recommendation': 'PR',
+    'Proposed Edited Recommendation': 'PER',
+    'Recommendation': 'REC',
+    'Draft Note': 'DNOTE',
+    'Note': 'NOTE',
+    'Draft Registry': 'DRY',
+    'Candidate Registry Draft': 'CRYD',
+    'Candidate Registry': 'CRY',
+    'Registry': 'RY',
+    'Statement': 'STMT'
 };
+function getStatus(version, versions) {
+    if (version.status in STATUSES) {
+        return STATUSES[version.status];
+    }
+    if (version.status === 'Retired') {
+        // Let's look for something that looks like a W3C status in the URL of
+        // the retired spec and fallback to the status of the first previous
+        // version for which we can determine a non-retired status otherwise.
+        // Note: while that may seem strange, more than one "Retired" version
+        // of a spec may be published for a single spec. For an example, see:
+        // https://api.w3.org/specifications/cselection/versions?embed=1
```

Comment:
Thanks for the exploration! It surprised me that we could publish 2 "Retired" versions in a row - and I see that surprised you as well ;) - but it's not a problem per se, could indeed be needed as in the case of the web-packaging specs, and is easy to handle in the end...

Reply:
I've updated the records and now, only web-packaging has more than 1 retired versions.
Note that we are looking at moving that 'retired' bit on the specification instead of having it on a specific version but if we do so, I'll make sure to retain the status 'Retired' on the version to avoid breaking things.